pkg/chains/formats/slsa/v2alpha2/internal/resolved_dependencies/resolved_dependencies.go

```diff
@@ -0,0 +1,361 @@
+/*
+Copyright 2023 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resolvedDependencies
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/common"
+	v1 "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v1"
+	"github.com/tektoncd/chains/internal/backport"
+	"github.com/tektoncd/chains/pkg/artifacts"
+	"github.com/tektoncd/chains/pkg/chains/formats/slsa/attest"
+	"github.com/tektoncd/chains/pkg/chains/formats/slsa/internal/material"
+	"github.com/tektoncd/chains/pkg/chains/objects"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"go.uber.org/zap"
+	"knative.dev/pkg/logging"
+)
+
+const (
+	pipelineConfigName     = "config/pipeline"
+	taskConfigName         = "config/task"
+	pipelineTaskConfigName = "config/pipelineTask"
+	containerImageName     = "image"
+	inputResultName        = "inputs/result"
+	pipelineResourceName   = "pipelineResource"
+)
+
+// TaskRunResolvedDependencies constructs `predicate.resolvedDependencies` section by collecting all the artifacts that influence a taskrun such as source code repo and step&sidecar base images.
+func TaskRunResolvedDependencies(ctx context.Context, tro *objects.TaskRunObject) (*[]v1.ResourceDescriptor, error) {
+	var resolvedDependencies []v1.ResourceDescriptor
+
+	// add top level task config
+	if p := tro.Status.Provenance; p != nil && p.RefSource != nil {
+		rd := v1.ResourceDescriptor{
+			Name:   taskConfigName,
+			URI:    p.RefSource.URI,
+			Digest: p.RefSource.Digest,
+		}
+		resolvedDependencies = append(resolvedDependencies, rd)
+	}
+
+	// add step images
+	if err := addStepImagesToResolvedDependencies(tro.Status.Steps, &resolvedDependencies); err != nil {
+		return &resolvedDependencies, err
```

Comment:
if there is an error in any step, shouldn't we return a nil resolvedDependencies instead of an incomplete list?